src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -726,27 +726,34 @@ private async Task<List<LaunchCommand>> GetInitializeCommands()
 
                     // check for remote
                     string destination = localLaunchOptions?.MIDebuggerServerAddress;
+                    bool useExtendedRemote = localLaunchOptions?.UseExtendedRemote ?? false;
                     if (!string.IsNullOrWhiteSpace(destination))
                     {
-                        commands.Add(new LaunchCommand("-target-select remote " + destination, string.Format(CultureInfo.CurrentCulture, ResourceStrings.ConnectingMessage, destination)));
+                        string remoteMode = useExtendedRemote ? "extended-remote" : "remote";
+                        commands.Add(new LaunchCommand($"-target-select {remoteMode} {destination}", string.Format(CultureInfo.CurrentCulture, ResourceStrings.ConnectingMessage, destination)));
                     }
-                    else // gdbserver is already attached when using LocalLaunchOptions
+                    // Allow attach after connection only in extended-remote mode
+                    if (useExtendedRemote || (!useExtendedRemote && string.IsNullOrWhiteSpace(destination)))
                     {
-                        Action<string> failureHandler = (string miError) =>
+                        string processId = localLaunchOptions?.ProcessId.Value.ToString(CultureInfo.InvariantCulture);
+                        if (!string.IsNullOrWhiteSpace(processId))
```

Comment:
>if (!string.IsNullOrWhiteSpace(processId)) [](http://example.com/codeflow?start=24&length=42)

nit: it shouldn't be possible for `processId` to be null or whitespace

Reply:
Ah right, didn't notice that. Will change back.